app/models/spree/promotion_decorator.rb

```diff
@@ -0,0 +1,4 @@
+Spree::Promotion.class_eval do
```

Comment:
@davidwinter looked at doing this via a concern but adding in the relation looked nasty